ntbs-service/Middleware/AuditGetRequestMiddleware.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using ntbs_service.Services;
+
+namespace ntbs_service.Middleware
+{
+    public class AuditGetRequestMiddleWare
+    {
+        private readonly RequestDelegate _next;
+
+        public AuditGetRequestMiddleWare(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context, IAuditService auditService)
+        {
+            var request = context.Request;
+
+            if (request.Method == "GET" && request.Query.ContainsKey("id"))
+            {
+                var id = int.Parse(request.Query["id"].ToString());
+                var modelName = request.Path.Value.Split("/").Last();
+                await auditService.OnGetAuditAsync(id, modelName, viewType: "Full");

```

Comment:
Should probably add a "TODO" here to say that the viewType could also be Cluster in future, and will be determined from the request path eventually